src/Passes/Source/Commandline/ParameterParser.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// Parameter parser class which allows the developer to specify a set of default settings and
+    /// update those using the commandline argc and argv.
+    class ParameterParser
+    {
+      public:
+        using String      = std::string;
+        using Arguments   = std::vector<String>;
+        using Flags       = std::unordered_set<String>;
+        using SettingsMap = std::unordered_map<String, String>;
+
+        // Construction and deconstrution configuration
+        //
+
+        /// Parameter parsers requires a setting class to store
+        /// parameters passed. The parameter parser takes a set of
+        /// default Settings as its first argument.
+        ParameterParser() = default;
```

Comment:
In the code I don't see the confirmation to "The parameter parser takes a set of default Settings as its first argument".

Reply:
You are right. This docstring is outdated as the defaults are derived from the `ConfigurationManager`.